worker2/taskcounter.go

```diff
@@ -0,0 +1,82 @@
+package worker
+
+import (
+	"sync"
+	"time"
+
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+)
+
+// taskCounter keeps count of number of active tasks as well as pending time.
+type taskCounter struct {
+	m         sync.Mutex
+	c         sync.Cond
```

Comment:
@petemoore, you are very right indeed... this is NOT idiomatic.

So I'm trying to do here is create a concurrency primitive that deals exclusive the flow that the worker has...
In this case `taskCounter` tracks the number of tasks currently running, and when this number is zero, the idleTimer is increasing.

That way you can always: 
 * Get the amount of idle time elapsed at a given moment
 * Get the number of tasks currently running
 * Wait for zero tasks to be running
 * Wait for less than N tasks to be running

I could build all these into the worker.... but I would effectively just inline the functions from this file... at different places.. That makes it really hard to see if I'm doing the right thing.
My thinking is that this makes it easier to understand... because I'm splitting out some of the concurrency logic, separating it from the logic that does polling, etc...

Am I all wrong? I fear that if I tried to use channels to track this kind of counter and support the 4 operations listed above it would become hugely complicated.